File: src/main/scala/de/mineformers/allomancy/block/properties/PropertyString.scala
package de.mineformers.allomancy.block.properties

import net.minecraft.block.properties.PropertyHelper

/**
  * PropertyString
  *
  * @author PaleoCrafter
  */
class PropertyString(name: String, allowedValues: String*) extends PropertyHelper[String](name, classOf[String]) {
  import scala.collection.JavaConverters._

  private val jValues = allowedValues.asJavaCollection

  override def getName(value: String): String = value

  override def getAllowedValues = jValues
}

File: src/main/java/de/mineformers/allomancy/block/properties/PropertyString.java
package de.mineformers.allomancy.block.properties;

import net.minecraft.block.properties.PropertyHelper;

import java.util.Arrays;
import java.util.Collection;

/**
 * PropertyString
 *
 * @author PaleoCrafter
 */
public class PropertyString extends PropertyHelper<String> {
    private Collection<String> allowedValues;

    public PropertyString(String name, String... allowedValues) {
        super(name, String.class);
        this.allowedValues = Arrays.asList(allowedValues);
    }

    @Override
    public String getName(String value) {
        return value;
    }

    @Override
    public Collection<String> getAllowedValues() {
        return allowedValues;
    }
}
